### ISSUE TITLE
Compilation error: Undefined reference to __atomic_fetch_add_8

### ISSUE THREAD

**Issue by mineas312** [microsoft/vcpkg#22843]
**Describe the bug**
Compilation error.

**Environment**
- OS: Raspbian
- Compiler: g++ (Raspbian 10.2.1-6+rpi1)

**To Reproduce**
1. ./bootstrap-vcpkg.sh

**Expected behavior**
Compiles.

**Failure logs** 
```
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o: in function `vcpkg::Filesystem::write_contents(vcpkg::Path const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, vcpkg::LineInfo)':
files.cpp:(.text+0x1fd0): undefined reference to `__atomic_fetch_add_8'
```
I do not know where full error log is located.  
There are other errors like this and I think it basically does not link a library related with this functions.

**Comment by sandeepmistry** [comment]
I'm also facing this issue:

```
cSt11char_traitsIcESaIcEEERSt10error_code]+0xe0): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: files.cpp:(.text._ZN5vcpkg14RealFilesystem23write_contents_and_dirsERKNS_4PathERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSt10error_code[_ZN5vcpkg14RealFilesystem23write_contents_and_dirsERKNS_4PathERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSt10error_code]+0x27c): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o:files.cpp:(.text._ZN5vcpkg14RealFilesystem22copy_regular_recursiveERKNS_4PathES3_RSt10error_code[_ZN5vcpkg14RealFilesystem22copy_regular_recursiveERKNS_4PathES3_RSt10error_code]+0x80): more undefined references to `__atomic_fetch_add_8' follow
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o: in function `vcpkg::Json::Reader::get_reader_stats()':
json.cpp:(.text+0x1754): undefined reference to `__atomic_load_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o: in function `vcpkg::Json::get_json_parsing_stats()':
json.cpp:(.text+0x1bb8): undefined reference to `__atomic_load_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o: in function `vcpkg::Json::parse(vcpkg::StringView, vcpkg::StringView)':
json.cpp:(.text+0x8004): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o: in function `vcpkg::Json::parse_file(vcpkg::Filesystem const&, vcpkg::Path const&, std::error_code&)':
json.cpp:(.text+0x828c): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o: in function `vcpkg::Json::parse_file(vcpkg::LineInfo, vcpkg::Filesystem const&, vcpkg::Path const&)':
json.cpp:(.text+0xa7d8): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/sourceparagraph.cpp.o: in function `vcpkg::parse_manifest_configuration[abi:cxx11](vcpkg::StringView, vcpkg::Json::Object const&)':
sourceparagraph.cpp:(.text+0x51ac): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/sourceparagraph.cpp.o: in function `vcpkg::SourceControlFile::parse_manifest_object(vcpkg::StringView, vcpkg::Json::Object const&)':
sourceparagraph.cpp:(.text+0xbf74): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/vcpkgpaths.cpp.o:vcpkgpaths.cpp:(.text+0x9068): more undefined references to `__atomic_fetch_add_8' follow
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/system.process.cpp.o: in function `vcpkg::get_subproccess_stats()':
system.process.cpp:(.text+0x18fc): undefined reference to `__atomic_load_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/system.process.cpp.o: in function `vcpkg::cmd_execute_and_stream_data(vcpkg::Command const&, vcpkg::InWorkingDirectory, std::function<void (vcpkg::StringView)>, vcpkg::Environment const&)':
system.process.cpp:(.text+0x1e4c): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/system.process.cpp.o: in function `vcpkg::cmd_execute(vcpkg::Command const&, vcpkg::InWorkingDirectory, vcpkg::Environment const&)':
system.process.cpp:(.text+0x38fc): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/paragraphs.cpp.o: in function `vcpkg::Paragraphs::get_load_ports_stats()':
paragraphs.cpp:(.text+0x2070): undefined reference to `__atomic_load_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/paragraphs.cpp.o: in function `vcpkg::Paragraphs::try_load_port(vcpkg::Filesystem const&, vcpkg::Path const&)':
paragraphs.cpp:(.text+0x407c): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/paragraphs.cpp.o: in function `vcpkg::Paragraphs::try_load_cached_package[abi:cxx11](vcpkg::Filesystem const&, vcpkg::Path const&, vcpkg::PackageSpec const&)':
paragraphs.cpp:(.text+0x6494): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/paragraphs.cpp.o: in function `vcpkg::Paragraphs::try_load_port_text(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, vcpkg::StringView, bool)':
paragraphs.cpp:(.text+0x6730): undefined reference to `__atomic_fetch_add_8'
collect2: error: ld returned 1 exit status
[133/133] Linking CXX executable vcpkg-test
FAILED: vcpkg-test 
: && /usr/bin/c++ -O3 -DNDEBUG  CMakeFiles/vcpkglib.dir/src/vcpkg/base/checks.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/chrono.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/cofffilereader.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/downloads.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/hash.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/messages.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/parse.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/strings.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/stringview.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/system.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/system.print.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/system.process.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/system.proxy.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/uint128.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/unicode.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/base/xmlserializer.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/archives.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/binarycaching.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/binaryparagraph.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/build.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/buildenvironment.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/cmakevars.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.activate.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.add-version.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.add.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.autocomplete.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.buildexternal.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.cache.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.check-support.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.ci.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.ciclean.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.civerifyversions.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.contact.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.create.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.dependinfo.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.edit.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.env.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.fetch.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.find.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.format-manifest.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.generate-message-map.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.hash.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.info.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.init-registry.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.integrate.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.list.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.new.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.owns.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.porthistory.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.portsdiff.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.regenerate.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.search.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.setinstalled.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.upgrade.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.upload-metrics.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.use.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.version.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.xdownload.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.xvsinstances.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.zbootstrap-standalone.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.zce.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/commands.zprintconfig.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/configuration.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/configure-environment.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/dependencies.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/export.chocolatey.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/export.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/export.ifw.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/export.prefab.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/globalstate.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/help.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/input.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/install.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/metrics.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/packagespec.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/paragraphs.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/platform-expression.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/portfileprovider.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/postbuildlint.buildtype.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/postbuildlint.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/registries.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/remove.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/sourceparagraph.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/statusparagraph.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/statusparagraphs.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/tools.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/triplet.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/update.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/userconfig.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/vcpkgcmdarguments.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/vcpkglib.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/vcpkgpaths.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/versiondeserializers.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/versions.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/versiont.cpp.o CMakeFiles/vcpkglib.dir/src/vcpkg/visualstudio.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/arguments.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/binarycaching.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/catch.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/chrono.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/commands.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/configmetadata.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/configparser.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/dependencies.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/downloads.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/files.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/hash.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/json.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/manifests.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/mockcmakevarsprovider.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/optional.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/paragraph.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/plan.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/platform-expression.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/registries.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/specifier.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/statusparagraphs.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/strings.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/stringview.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/system.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/uint128.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/update.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/util.cpp.o CMakeFiles/vcpkg-test.dir/src/vcpkg-test/versionplan.cpp.o -o vcpkg-test  _deps/fmt-build/libfmt.a  -pthread && :
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o: in function `vcpkg::Filesystem::write_contents(vcpkg::Path const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, vcpkg::LineInfo)':
files.cpp:(.text+0x1fd0): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o: in function `vcpkg::Filesystem::write_contents_and_dirs(vcpkg::Path const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, vcpkg::LineInfo)':
files.cpp:(.text+0x2190): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: files.cpp:(.text+0x23d8): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o: in function `vcpkg::Filesystem::copy_regular_recursive(vcpkg::Path const&, vcpkg::Path const&, vcpkg::LineInfo)':
files.cpp:(.text+0x5604): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o: in function `vcpkg::get_filesystem_stats()':
files.cpp:(.text+0x59a4): undefined reference to `__atomic_load_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o: in function `vcpkg::RealFilesystem::write_contents(vcpkg::Path const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::error_code&)':
files.cpp:(.text._ZN5vcpkg14RealFilesystem14write_contentsERKNS_4PathERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSt10error_code[_ZN5vcpkg14RealFilesystem14write_contentsERKNS_4PathERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSt10error_code]+0xdc): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o: in function `vcpkg::StatsTimer::~StatsTimer()':
files.cpp:(.text._ZN5vcpkg10StatsTimerD2Ev[_ZN5vcpkg10StatsTimerD5Ev]+0x50): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o: in function `vcpkg::RealFilesystem::create_directories(vcpkg::Path const&, std::error_code&)':
files.cpp:(.text._ZN5vcpkg14RealFilesystem18create_directoriesERKNS_4PathERSt10error_code[_ZN5vcpkg14RealFilesystem18create_directoriesERKNS_4PathERSt10error_code]+0x278): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o: in function `vcpkg::RealFilesystem::write_contents_and_dirs(vcpkg::Path const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::error_code&)':
files.cpp:(.text._ZN5vcpkg14RealFilesystem23write_contents_and_dirsERKNS_4PathERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSt10error_code[_ZN5vcpkg14RealFilesystem23write_contents_and_dirsERKNS_4PathERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSt10error_code]+0xe0): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: files.cpp:(.text._ZN5vcpkg14RealFilesystem23write_contents_and_dirsERKNS_4PathERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSt10error_code[_ZN5vcpkg14RealFilesystem23write_contents_and_dirsERKNS_4PathERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSt10error_code]+0x27c): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o:files.cpp:(.text._ZN5vcpkg14RealFilesystem22copy_regular_recursiveERKNS_4PathES3_RSt10error_code[_ZN5vcpkg14RealFilesystem22copy_regular_recursiveERKNS_4PathES3_RSt10error_code]+0x80): more undefined references to `__atomic_fetch_add_8' follow
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o: in function `vcpkg::Json::Reader::get_reader_stats()':
json.cpp:(.text+0x1754): undefined reference to `__atomic_load_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o: in function `vcpkg::Json::get_json_parsing_stats()':
json.cpp:(.text+0x1bb8): undefined reference to `__atomic_load_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o: in function `vcpkg::Json::parse(vcpkg::StringView, vcpkg::StringView)':
json.cpp:(.text+0x8004): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o: in function `vcpkg::Json::parse_file(vcpkg::Filesystem const&, vcpkg::Path const&, std::error_code&)':
json.cpp:(.text+0x828c): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/json.cpp.o: in function `vcpkg::Json::parse_file(vcpkg::LineInfo, vcpkg::Filesystem const&, vcpkg::Path const&)':
json.cpp:(.text+0xa7d8): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/sourceparagraph.cpp.o: in function `vcpkg::parse_manifest_configuration[abi:cxx11](vcpkg::StringView, vcpkg::Json::Object const&)':
sourceparagraph.cpp:(.text+0x51ac): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/sourceparagraph.cpp.o: in function `vcpkg::SourceControlFile::parse_manifest_object(vcpkg::StringView, vcpkg::Json::Object const&)':
sourceparagraph.cpp:(.text+0xbf74): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/vcpkgpaths.cpp.o:vcpkgpaths.cpp:(.text+0x9068): more undefined references to `__atomic_fetch_add_8' follow
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/system.process.cpp.o: in function `vcpkg::get_subproccess_stats()':
system.process.cpp:(.text+0x18fc): undefined reference to `__atomic_load_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/system.process.cpp.o: in function `vcpkg::cmd_execute_and_stream_data(vcpkg::Command const&, vcpkg::InWorkingDirectory, std::function<void (vcpkg::StringView)>, vcpkg::Environment const&)':
system.process.cpp:(.text+0x1e4c): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/base/system.process.cpp.o: in function `vcpkg::cmd_execute(vcpkg::Command const&, vcpkg::InWorkingDirectory, vcpkg::Environment const&)':
system.process.cpp:(.text+0x38fc): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/paragraphs.cpp.o: in function `vcpkg::Paragraphs::get_load_ports_stats()':
paragraphs.cpp:(.text+0x2070): undefined reference to `__atomic_load_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/paragraphs.cpp.o: in function `vcpkg::Paragraphs::try_load_port(vcpkg::Filesystem const&, vcpkg::Path const&)':
paragraphs.cpp:(.text+0x407c): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/paragraphs.cpp.o: in function `vcpkg::Paragraphs::try_load_cached_package[abi:cxx11](vcpkg::Filesystem const&, vcpkg::Path const&, vcpkg::PackageSpec const&)':
paragraphs.cpp:(.text+0x6494): undefined reference to `__atomic_fetch_add_8'
/usr/bin/ld: CMakeFiles/vcpkglib.dir/src/vcpkg/paragraphs.cpp.o: in function `vcpkg::Paragraphs::try_load_port_text(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, vcpkg::StringView, bool)':
paragraphs.cpp:(.text+0x6730): undefined reference to `__atomic_fetch_add_8'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Comment by mathisloge** [comment]
seems like you need to add `-latomic` when building vcpkg with raspbian gcc.

**Comment by clarkjs0817** [comment]
I wanna ask which CML should target_link_libraries(tgt PRIVATE atomic) added into ?

**Comment by mathisloge** [comment]
Propably as a public lib in https://github.com/microsoft/vcpkg-tool/blob/f95f306ecdd7f9228aa8ea22ccae53f09e22c630/CMakeLists.txt#L145 so that vcpkg-tool gets it, too. 
e.g. here https://github.com/microsoft/vcpkg-tool/blob/f95f306ecdd7f9228aa8ea22ccae53f09e22c630/CMakeLists.txt#L183

**Comment by clarkjs0817** [comment]
Thanks for your reply, I still have some problems to trouble. I have same mistakes as issues above,my information is as follows:

and I did as&nbsp;https://gitlab.kitware.com/cmake/cmake/-/issues/21174&nbsp; told me. I added&nbsp;-DCMAKE_EXE_LINKER_FLAGS="-latomic" as follows :

but the questions haven't been solved. It still has the same error as before.


Could you give me some tips about it ?&nbsp; &nbsp;Thanks for your help.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/vcpkg"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年1月30日(星期天) 晚上11:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/vcpkg] Compilation error: Undefined reference to __atomic_fetch_add_8 (Issue #22843)





 
Propably as a public lib in https://github.com/microsoft/vcpkg-tool/blob/f95f306ecdd7f9228aa8ea22ccae53f09e22c630/CMakeLists.txt#L145 so that vcpkg-tool gets it, too.
 e.g. here https://github.com/microsoft/vcpkg-tool/blob/f95f306ecdd7f9228aa8ea22ccae53f09e22c630/CMakeLists.txt#L183
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by hifiberry** [comment]
Is there a workaround? Editing the file locally doesn't work as the bootstrap overwrites it again.

**Comment by mprather** [comment]
@mineas312, @sandeepmistry, @hifiberry - a fix was committed to the rep on 3/25 which addresses the linker issue. The fix to vcpkg-tool was tested with Raspberry Pi OS 10, gcc 8.3, and a 32-bit board (3 B+). This plus another fix for #21142 has unblocked RPi compilation of vcpkg (as best I can tell). If you are still having issues, might be worth updating this issue with new info.

**Comment by sandeepmistry** [comment]
Hi @mprather, thank you for the update. I can confirm building (with commit 27fd32e91c172e8b8a2ee338efc088a0c0103348) on a Raspberry Pi 4 is now successful.
